plotting-visualization/rtdose_visual.py

```diff
@@ -0,0 +1,250 @@
+
+    """
+   
+    fig = event.canvas.figure
+    ax = fig.axes[0]
+    
+    if event.key == 'p':
+        previous_slice(ax)
+        draw(ax,False)
+    elif event.key == 'n':
+        next_slice(ax)
+        draw(ax,False)
+    elif event.key == 'c':
+        change_axis(ax) 
+        draw(ax,True)
+
+    fig.canvas.draw_idle()
+
+def draw(ax,changing):
+    """
+    Draws the new plot once the key input has changed the settings 
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+
+    changing : Boolean
+        A boolean variable that remains False unless we change the axis we are plotting.
+
+    Returns
+    -------
+    None.
+
+    """
+    volume= ax.volume
+    if ax.axis== 0:
+        if changing:
+            del ax.images[0] #this is needed in order to avoid making a new plot in ax.images[1]
+            ax.imshow(volume[ax.index,:,:],cMap, vmin=cMin, vmax=cMax)
+        else:
+            ax.images[0].set_array(volume[ax.index,:,:])
+       
+    elif ax.axis== 1:
+        if changing:
+            del ax.images[0]
+            ax.imshow(volume[:,ax.index,:],cMap, vmin=cMin, vmax=cMax)
+        else:
+            ax.images[0].set_array(volume[:,ax.index,:])
+    
+    elif ax.axis == 2:
+         if changing:
+            del ax.images[0]             
+            ax.imshow(volume[:,:,ax.index],cMap, vmin=cMin, vmax=cMax)
+         else:
+            ax.images[0].set_array(volume[:,:,ax.index])
+       
+
+def previous_slice(ax):
+    """
+    Go to the previous slice.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+        
+    Returns
+    -------
+    None.
+
+    """
+   
+    volume = ax.volume
+    ax.index = (ax.index - 1) % volume.shape[ax.axis]  # wrap around using %
+    
+def next_slice(ax):
+    """
+    Go to the next slice.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+
+    Returns
+    -------
+    None.
+
+    """
+    volume = ax.volume
+    ax.index = (ax.index + 1) % volume.shape[ax.axis]
+
+def change_axis(ax):
+    """
+    Changes the axis of view
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+    Returns
+    -------
+    None.
+
+    """
+    volume = ax.volume
+    ax.axis = (ax.axis + 1) % 3
+    ax.index = volume.shape[ax.axis] // 2
+
+# Define the path folder where the file is , or directly the RTdose full file name
+path = "RD.1.2.246.352.71.7.2088656855.452097.20110920152341.dcm"
+global cMin, cMax,cMap
+cMap = 'turbo'
+cMin = None
+cMax = None
+
+
+# This would print all the files and directories
+
+if os.path.isdir(path):
+    
+    dirs = os.listdir( path )
+    for file in dirs:
+       ds = pydicom.dcmread(path+"/"+file)
+       if ds.file_meta.MediaStorageSOPClassUID == '1.2.840.10008.5.1.4.1.1.481.2':
```

Comment:
Here `file` could be itself be a directory, leading to an error.  Could trap the exception or embed in an `if not os.path.isdir(file):`...

Reply:
Similarly, without the `force=True` passed to `dcmread`, this will bail with an error on any file that it not actually DICOM, or is old-style DICOM without proper preamble, etc. (like pydicom's rtdose.dcm test file).

Reply:
Done